### ISSUE TITLE
Unknown error in macs2 step

### ISSUE THREAD

**Issue by adamtongji** [kundajelab/atac_dnase_pipelines#57]
Hi,
When I ran two SE ATAC-seq  fastq files with following commands:
```
bds /home/adam/new_dir/atac_dnase_pipelines/atac.bds -species hg19 -nth 16 -fastq1 treat1.fastq -fastq2 treat2.fastq -se -trimmed_fastq -mem_dedup 30G -memory 30G -ENCODE3 -subsample_xcor 1M 
```

The following error reports:
__/Data/adam/test_atac/data/atac.bds.20170708_144320_377/task.callpeak_macs2_atac.macs2_atac_rep1_pr1.pval0.1.line_74.id_12.sh: line 61: syntax error near unexpected token `;;'__

All MACS2 commands failed with this same error. I ran with two PE bam files, and same error happens.  I have been struggling with this bug for a whole day, but still could not find the correct solution.  Could help me fix it?

Great thanks!

**Comment by chrisprobert** [comment]
Hi Adam - this was a regression that I accidentally introduced this week with #56 -- I've just fixed it in 759f19e. Please try fetching from master, and let us know if this fixes it.

cc: @leepc12

**Comment by adamtongji** [comment]
Hello,
Thanks for your quick reply! 
 
But #56 may not fix for my problem.  I checked the "modules/env.bds" file, and the "atac_dnase_pipeline" on my server has already been the update version with your latest revision —— with "sleep $cluster_task_delay;" in line 116th.  

I ran the pipe on a single centos7 workstation rather than a SGE cluster. This workstation has already been installed with most bioinformatics tools, and some of them may be the different version of tools in bds_atac conda environment.  I checked main software like macs2, samtools and bowtie2 in bds_atac conda env, and their absolute paths are correctly located conda env.  I am not sure if I should find a "clean" linux server for installation.

The context of my error is below:

> INFO  @ Sat, 08 Jul 2017 19:45:59:  .4 Write summits bed file... 
 > /Data/adam/test_atac/PE_bam_data/out/peak/macs2/pooled_pseudo_reps/ppr1/limb_11.5_day_rep1.nodup.pr1.tn5_limb_11.5_day_rep2.nodup.pr1.tn5.pf.pval0.1_summits.bed 		
> INFO  @ Sat, 08 Jul 2017 19:46:06: Done! 	
>
> /Data/adam/test_atac/PE_bam_data/atac.bds.20170708_190245_369/task.callpeak_macs2_atac. 
 > macs2_atac_ppr1.pval0.1.line_74.id_29.sh: line 61: syntax error near unexpected token `;;'
>
> Fatal error: /home/adam/new_dir/atac_dnase_pipelines/atac.bds, line 1115, pos 2. Task/s failed.


Thanks for your assistance!

**Comment by chrisprobert** [comment]
Apologies if this wasn't clear, but I actually meant that #56 *caused* the
problem; it should be fixed in the commit I pushed to master today:
https://github.com/kundajelab/atac_dnase_pipelines/commit/759f19eb9494f7376ec671967430591a45718a04

**Comment by adamtongji** [comment]
It works for my previous error!  Sorry for my mistake in checking the wrong commit of #56. 

There is a new small bug as following:
>Chromosome chr4_GL456216_random isn't in /Data/adam/atac_dnase_pipelines/bds_pipeline_genome_data/mm10/mm10.chrom.sizes line 4783025 of stdin

I ran the pipeline with ATAC-seq PE bam from ENCODE portal, and the genome version is mm10.  I guess this bug is caused by the difference of mm10 chrom.size file in bds_pipeline_genome_data folder.  In this case, the chromosome name in  bds_pipeline_genome_data folder is "GL456216.1".  

Thanks!

**Comment by adamtongji** [comment]
A new bug in centos7 as following:

> bgzip: error while loading shared libraries: libbz2.so.1.0: cannot open shared object file: No such file or directory
> tabix: error while loading shared libraries: libbz2.so.1.0: cannot open shared object file: No such file or directory

A possible solution I search from google:
```
sudo yum install bzip2-devel
sudo ln -s `find /usr/lib64/ -type f -name "libbz2.so.1*"` /usr/lib64/libbz2.so.1.0
```
thanks~

**Comment by leepc12** [comment]
1) ;; syntax problem is fixed in the commit ffb76c34b781eb2831d7dedb231fedb128c41040 in the master branch. Chris, sorry I made another commit for this fix before checking your PR.

2) About the GL456216.1 issue, you are right. You can pipelines from FASTQS so that the reference genome used for mapping FASTQS is consistent with chr. sizes file in pipeline's builtin genome database for mm10. Or you can manually specify your own chr. sizes file with `-chrsz [YOUR_CHRSZ_FILE_PATH]`.

3) Thanks for the solution for libbz2 issue, it's already been discussed here (https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!msg/klab_genomic_pipelines_discuss/HTsdB77QcbM/bdDWiT39AAAJ). I added your solution for CentOS to this channel.

**Comment by rfarouni** [comment]
hi @leepc12 

I also have the libbz2 issue, but since I don't have sudo privileges on my cluster I am not able to sudo yum install or change the symbolic links on the system. Is there another way to solve the problem?

**Comment by leepc12** [comment]
For those who don't have super-user privileges on the system, I added `-no_browser_tracks` flag (commit 	4c02919) to disable browser track generation, which needs bzip2 shared libraries.

**Comment by leepc12** [comment]
`libbz2.so.1.0` issue has been finally fixed in the commit cba7cc6. I lowered version of htslib (tabix and bgzip).

1) If you are working on Stanford clusters (SCG4 or Sherlock), you don't need any further action.

2) On other computers, please uninstall and re-install dependencies.
```
$ bash uninstall_dependencies.sh
$ bash install_dependencies.sh
```
